nobl9/resource_slo.go

```diff
@@ -727,6 +752,24 @@ func unmarshalObjectives(d *schema.ResourceData, spec map[string]interface{}) er
 	return d.Set("objective", schema.NewSet(objectiveHash, objectivesTF))
 }
 
+func objectiveHash(objective interface{}) int {
```

Comment:
by hashing the whole object instead of just a few immutable fields we will always have a difference in the whole when changing anything in the `objective`. So it will look sth like:

```
- objective {
-   name = "test"
-   value = 0.1
-  }

+ objective {
+   name = "test
+   value = 0.2
+  }
```

instead of just

```
objective {
    name = "test
+  value = 0.2
 }
```

Shouldn't we just hash the immutable fields? maybe just the new `objective.name` field? I know it can be changed once but I think this would be reasonable compromise.

Reply:
According to this topic [https://discuss.hashicorp.com/t/typeset-is-not-picking-up-correct-changes/23735/2](https://discuss.hashicorp.com/t/typeset-is-not-picking-up-correct-changes/23735/2) this is how TF handles diff for nested TypeSet. We can handle it with Update func as suggested, but I think this is a bit out of scope here.

Reply:
Ok, agreed, this is out of scope then